### PR TITLE
vimPluginsUpdater: add missing nix-prefetch-git dependency

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

I identified some issues with the new `vimPluginsUpdater` tool.

- First, the derivation was lacking the `nix-prefetch-git` dependency. (I provide a fix for this one)
- Second, It seems that the path to the treesitter `lockfile.json` [is wrongly provided](https://github.com/NixOS/nixpkgs/blob/b84b384401e403d99ff999d63b4c52c2c0ce3667/pkgs/applications/editors/vim/plugins/update.py#L132). I don't really know what was intended here, so I might let you fix it.

cc @teto @figsoda 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Error message for the second bug:
```
path is '/nix/store/9gcnfrxl44gjvdfsrjkgd3q1qln5rzmd-7fd86b7164442d3b5ec2c81b2694d040e716b5cf.tar.gz'
committing to nixpkgs "vimPlugins: update"
1344 plugins were checked
updated pkgs/applications/editors/vim/plugins/generated.nix
no changes in working tree to commit
updating nvim-treesitter grammars
Traceback (most recent call last):
  File "/nix/store/5mzm2wv4vvp3c7a7i2rgh5cjybadds1w-vim-plugins-updater-0.1/bin/..vim-plugins-updater-wrapped-wrapped", line 162, in <module>
    main()
  File "/nix/store/5mzm2wv4vvp3c7a7i2rgh5cjybadds1w-vim-plugins-updater-0.1/bin/..vim-plugins-updater-wrapped-wrapped", line 158, in main
    editor.run()
  File "/nix/store/5mzm2wv4vvp3c7a7i2rgh5cjybadds1w-vim-plugins-updater-0.1/lib/pluginupdate.py", line 560, in run
    getattr(self, command)(args)
  File "/nix/store/5mzm2wv4vvp3c7a7i2rgh5cjybadds1w-vim-plugins-updater-0.1/bin/..vim-plugins-updater-wrapped-wrapped", line 134, in update
    lockfile = json.load(open(args.nixpkgs.join(os.environ["NVIM_TREESITTER"], "lockfile.json")))
  File "/nix/store/pzf6dnxg8gf04xazzjdwarm7s03cbrgz-python3-3.10.12/lib/python3.10/os.py", line 680, in __getitem__
    raise KeyError(key) from None
KeyError: 'NVIM_TREESITTER
```

**Comment by GaetanLepage** [comment]
Sorry, this is the actual error message:
```
committing to nixpkgs "vimPlugins: update"
path is '/nix/store/h5zi4ir02bfbmwgq11lf158n7ynj5fs2-4724694bc03ce1148860a46d9d77c3664d8188ab.tar.gz'
1344 plugins were checked
updated pkgs/applications/editors/vim/plugins/generated.nix
committing to nixpkgs "vimPlugins: resolve github repository redirects"
updating nvim-treesitter grammars
Traceback (most recent call last):
  File "/nix/store/n0x2paymmm4qlrg6c13adkh4hjn307hz-vim-plugins-updater-0.1/bin/..vim-plugins-updater-wrapped-wrapped", line 162, in <module>
    main()
  File "/nix/store/n0x2paymmm4qlrg6c13adkh4hjn307hz-vim-plugins-updater-0.1/bin/..vim-plugins-updater-wrapped-wrapped", line 158, in main
    editor.run()
  File "/nix/store/n0x2paymmm4qlrg6c13adkh4hjn307hz-vim-plugins-updater-0.1/lib/pluginupdate.py", line 560, in run
    getattr(self, command)(args)
  File "/nix/store/n0x2paymmm4qlrg6c13adkh4hjn307hz-vim-plugins-updater-0.1/bin/..vim-plugins-updater-wrapped-wrapped", line 134, in update
    lockfile = json.load(open(args.nixpkgs.join(NVIM_TREESITTER_GENERATED_FILE, "lockfile.json")))
NameError: name 'NVIM_TREESITTER_GENERATED_FILE' is not defined. Did you mean: 'NVIM_TREESITTER_GENERATED_NIX'?
```

**Comment by GaetanLepage** [comment]
Ok, I drafted a fix.

File: pkgs/applications/editors/vim/plugins/update.py [hunk 1]
```diff
@@ -129,9 +131,12 @@ def update(self, args):
         if self.nvim_treesitter_updated:
             print("updating nvim-treesitter grammars")
             nvim_treesitter_dir = ROOT.joinpath("nvim-treesitter")
-            lockfile = json.load(open(args.nixpkgs.join(NVIM_TREESITTER_GENERATED_FILE, "lockfile.json")))
+            lockfile = json.load(open(os.path.join(os.environ['NVIM_TREESITTER'], "lockfile.json")))
```
Comment (teto) [inline pkgs/applications/editors/vim/plugins/update.py:134]: the goal of the last update was to decouple the updater from nixpkgs, so you could run it on arbitrary input files so here instead I would use `self.nixpkgs_repo` to find the root of the checkout

File: pkgs/applications/editors/vim/plugins/updater.nix [hunk 2]
```diff
@@ -38,7 +40,11 @@ buildPythonApplication {
     cp ${../../../../../maintainers/scripts/pluginupdate.py} $out/lib/pluginupdate.py
 
     # wrap python scripts
-    makeWrapperArgs+=( --prefix PATH : "${lib.makeBinPath [ nix my_neovim ]}" --prefix PYTHONPATH : "$out/lib" )
+    makeWrapperArgs+=(
+      --prefix PATH : "${lib.makeBinPath [ nix nurl nix-prefetch-git my_neovim ]}" \
+      --prefix PYTHONPATH : "$out/lib" \
+      --prefix NVIM_TREESITTER : "${vimPlugins.nvim-treesitter}" \
```
Comment (teto) [inline pkgs/applications/editors/vim/plugins/updater.nix:46]: I think you can remove this line as the nixpkgs path should be retreived from the parser
```
        common.add_argument(
            "--nixpkgs",
            type=str,
            default=os.getcwd(),
            help="Adjust log level",
        )
```

File: pkgs/applications/editors/vim/plugins/update.py (hunk 1 above)
Comment (GaetanLepage) [reply pkgs/applications/editors/vim/plugins/update.py:134]: Here we are not looking for the path of the repo but the nvim-treesitter derivation path right ? 
I.e. `/nix/store/xxxxxxx-nvim-treesitter/lockfile.json`

**Comment by GaetanLepage** [comment]
@teto Actually, feel free to take over this branch. I was trying to solve this myself, however, you might have a clearer idea on how this script should work

**Comment by teto** [comment]
cool I actually started to do that just 10mn ago xD

**Comment by teto** [comment]
ty for the work, I've opened a follow up at https://github.com/NixOS/nixpkgs/pull/260096

**Comment by GaetanLepage** [comment]
> ty for the work, I've opened a follow up at #260096

Great ! Thank you :)

**Pull request closed without merging**